src/compiler/emit_x86.bas

```diff
@@ -4153,8 +4153,30 @@ private sub hCMPI _
 		ostr = "test " + dst + COMMA + dst
 		outp ostr
 	else
-		ostr = "cmp " + dst + COMMA + src
-		outp ostr
+
+		'' immediate and offset? it's invalid for CMP so use a temp reg
+		if( (svreg->typ = IR_VREGTYPE_IMM) and (dvreg->typ = IR_VREGTYPE_OFS) ) then
+
+			dim reg as integer, isfree as integer, rname as string
```

Comment:
inner `rname` is shadowing the outer `rname` here, might be better to split into an extra function